pet-store-react-hooks/src/DeletePetModal/DeletePetModal.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+
+import { deletePetAsync } from '~infrastructure/api-client';
+import type { PetListItem } from '~infrastructure/api-types';
+import { ErrorMessage } from '~infrastructure/components/ErrorMessage/ErrorMessage';
+import { Modal } from '~infrastructure/components/Modal/Modal';
+import { formatDate } from '~infrastructure/utils';
+
+import '~infrastructure/components/Modal/Modal.css';
+
+import './DeletePetModal.css';
+
+type DeletePetModalProps = {
+  pet: PetListItem;
+  petKindsMap: Map<number, string>;
+  // Question: because im handling things differently on this callback from the parent component
+  // is it OK to use one callback with a parameter, or two - onClose() , onDelete()?
```

Comment:
Separate callbacks and you call both on success. One to notify the parent that the delete succeeded and one to close the modal.